File: Java/programas/PO-Abstracción/src/Cubo.java
//Declara clase regular derivada de Figuras.
public class Cubo extends Figuras {
    private double Lado; //Atributos privados.

    public Cubo(double lado){ //Constructor que inicializa atributo
        this.Lado = lado;
    }

    public double CalcularVolumen(){ //Función que regresa el volumen del cubo
        return Math.pow(Lado, 3);
    }

    public void DesplegarVolumen(){ //Procedimiento que despliega medida del lado y volumen del cubo
        System.out.println("\n---Datos del cubo---\n");
        System.out.println("La medida del lado del cubo es:" + Lado + "u.");
        System.out.println("El volumen del cubo es: " + CalcularVolumen() + "u^3.");
    }
}

File: C++/programas/POO-Abstraccion/POO-Abstraccion/Cubo.cpp
#define _USE_MATH_DEFINES //Define el uso de constantes matemáticas.
#include "Cubo.h" //Cabecera para trabajar con funciones matemáticas.
#include <iostream> //Cabecera para entrada y salida de datos.

//Inicialización de atributos por medio del constructor.
Cubo::Cubo(double lado) : Lado(lado){} //Forma alternativa de declarar un constructor.

//Declaración de métodos abstractos en clase derivada. Regresa el volumen de un cubo.
double Cubo::CalcularVolumen() { 
	return pow(Lado, 3);
}

//Declaración de métodos abstractos en clase derivada. Despliega datos de un cubo.
void Cubo::DesplegarVolumen() {
	std::cout << "\n--Datos del cubo--\n\n";
	std::cout << "La medida del lado del cubo es: " << Lado << "u.\n";
	std::cout << "El volumen del cubo es: " << CalcularVolumen() << "u^3.\n";
}